src/devices/tpms_eezrv.c

```diff
@@ -55,36 +55,50 @@ static int tpms_eezrv_decode(r_device *decoder, bitbuffer_t *bitbuffer)
     bitbuffer_invert(bitbuffer);
     pos = bitbuffer_search(bitbuffer, 0, pos, preamble_pattern, sizeof(preamble_pattern) * 8);
     if (pos >= bitbuffer->bits_per_row[0]) {
-        decoder_log(decoder, 2, __func__, "Preamble not found");
+        decoder_log(decoder, 3, __func__, "Preamble not found");
         return DECODE_ABORT_EARLY;
     }
-    if (pos + 7 * 8 > bitbuffer->bits_per_row[0]) {
+    if (pos + 8 * 8 > bitbuffer->bits_per_row[0]) {
         decoder_log(decoder, 2, __func__, "Length check fail");
         return DECODE_ABORT_LENGTH;
     }
-    uint8_t b[6]  = {0};
+    uint8_t b[7]  = {0};
     uint8_t cc[1] = {0};
     bitbuffer_extract_bytes(bitbuffer, 0, pos + 16, cc, sizeof(cc) * 8);
     bitbuffer_extract_bytes(bitbuffer, 0, pos + 24, b, sizeof(b) * 8);
+
     // verify checksum
-    if ((add_bytes(b, 6) & 0xff) != cc[0]) {
+    int computed_checksum = add_bytes(b, sizeof(b));
+    if (computed_checksum > 0xff) 
+        computed_checksum |= 0x80;
```

Comment:
Can you add a comment that the top bits fold around? That's what you mean here, right?
Did you observe a checksum bigger than 1FF ? I.e. what would 2FF do?

Reply:
If the checksum is greater than 0xFF then the MSB is set.  I occurs whether the bit is already set or not and was observed when checksum was in the 0x1FF and the 0x2FF range.  I did not observe a checksum higher than in the 0x200-0x2FF range.  I'm assuming this holds true for higher than 0x2FF, but I have not data to support it.